accelerator/apis.c

```diff
@@ -153,55 +153,153 @@ status_t api_find_transactions_obj_by_tag(
+                                      char const* const host,
+                                      uint16_t const port) {
+  serv->http.path = "/";
+  serv->http.content_type = "application/json";
+  serv->http.accept = "application/json";
+  serv->http.host = host;
+  serv->http.port = port;
+  serv->http.api_version = 1;
+  serv->http.ca_pem = NULL;
+  serv->serializer_type = SR_JSON;
+  iota_client_core_init(serv);
+}
+
+static transaction_array_t* _get_bundle_transactions(
+    iota_client_service_t* const serv, find_transactions_req_t* recv_req,
+    flex_trit_t const* const bundle_hash) {
+  flex_trit_t bundle_hash_flex[FLEX_TRIT_SIZE_243];
+  transaction_array_t* out_tx_objs = transaction_array_new();
+  recv_req->approvees = NULL;
+  recv_req->bundles = NULL;
+  recv_req->tags = NULL;
+  recv_req->addresses = NULL;
+
+  flex_trits_from_trytes(bundle_hash_flex, NUM_TRITS_BUNDLE, bundle_hash,
+                         NUM_TRITS_HASH, NUM_TRYTES_BUNDLE);
+  hash243_queue_push(&recv_req->bundles, bundle_hash);
+
+  retcode_t err =
+      iota_client_find_transaction_objects(serv, recv_req, out_tx_objs);
+  if (err != RC_OK) {
+    fprintf(
+        stderr,
+        "iota_client_find_transaction_objects failed with %d in retcode_t\n",
+        err);
+    return out_tx_objs;
+  }
+
+  return out_tx_objs;
+}
+
+static status_t _receive_bundle(char const* const host, uint16_t const port,
+                                tryte_t const* const bundle_hash,
+                                bundle_transactions_t* const bundle) {
+  iota_client_service_t serv;
+  find_transactions_req_t recv_req;
+  _recv_init_client_service(&serv, host, port);
+  iota_client_extended_init();
+
+  transaction_array_t* out_tx_objs =
+      _get_bundle_transactions(&serv, &recv_req, bundle_hash);
+
+  _get_first_bundle_from_transactions(out_tx_objs, bundle);
+
+  transaction_array_free(out_tx_objs);
+
+  iota_client_extended_destroy();
+  iota_client_core_destroy(&serv);
+
+  hash243_queue_free(&recv_req.bundles);
+  recv_req.bundles = NULL;
+  return SC_OK;
+}
+
 status_t api_receive_mam_message(const iota_client_service_t* const service,
-                                 const char* const obj, char** json_result) {
+                                 const char* const host, uint16_t port,
+                                 const char* const bundle_hash,
+                                 const char* const chid, char** json_result) {
+  mam_api_t api;
   status_t ret = SC_OK;
-  mam_api_t mam;
-
   tryte_t* payload_trytes = NULL;
   size_t payload_size = 0;
   bundle_transactions_t* bundle = NULL;
   bundle_transactions_new(&bundle);
   bool is_last_packet;
 
   // Creating MAM API
-  if (mam_api_init(&mam, (tryte_t*)SEED)) {
-    ret = SC_MAM_OOM;
-    goto done;
+  if (mam_api_init(&api, (tryte_t*)bundle_hash) != RC_OK) {
+    fprintf(stderr, "mam_api_init failed with err %d in retcode_t\n", ret);
```

Comment:
Don't `fprintf` here. If you want to logging, please use logger_helper like in `config.h`.